### PR TITLE
fix issue #65

### PR DESCRIPTION
**Pull request by KevRiver**
- [x] add main_test.go, /test/cases/00065-appy-config -> add test that triggers the same bug and fails
- [x] fix the bug -> wrap the line `len(*cfg.Skip) > 0 && len(cfg.Keep)` by nil pointer check statement
- [x] the test should now pass, commit

### CODE REVIEW

File: cmd/humanlog/main_test.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	"strings"
+	"testing"
+)
+
+func TestApplyConfigFromConfigFile_when_one_of_skip_or_keep_is_given(t *testing.T) {
+
+	wd, _ := os.Getwd()
+	dirs := strings.Split(wd, "/")
+	root := strings.Join(dirs[:len(dirs)-2], "/")
+	configFilePath := root + "/test/cases/00065-apply-config/config.json"
```
Comment (aybabtme) [inline cmd/humanlog/main_test.go:14]: can you switch this to use https://pkg.go.dev/testing#T.TempDir and write a file there?

you can use this struct: https://github.com/humanlogio/humanlog/blob/master/internal/pkg/config/config.go#L78-L98

maybe something like (pseudo-code):
```go
cfg := config.Config{
   Skip: []string{"foo", "bar"},
}

dir := t.TempDir()
f, err := os.TempFile(dir, "00065-apply-config.json)
require.NoError(t, err)

err = json.NewEncoder(f).Encode(cfg)
require.NoError(t, err)

err = f.Close()
require.NoError(t, err)

args := []string{"program-path", "--config", f.Name()}
...
```

File: cmd/humanlog/main_test.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import (
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestApplyConfigFromConfigFile_when_one_of_skip_or_keep_is_given(t *testing.T) {
+
+	wd, _ := os.Getwd()
+	dirs := strings.Split(wd, "/")
```
Comment (aybabtme) [inline cmd/humanlog/main_test.go:12]: fyi: when manipulating filepath, use https://pkg.go.dev/path/filepath since it's cross platform

**Comment by KevRiver** [comment]
[like]  강승곤 reacted to your message:
________________________________

**Comment by KevRiver** [comment]
got it! thanks for your review

iOS용 Outlook<https://aka.ms/o0ukef> 다운로드
________________________________
보낸 사람: Antoine Grondin ***@***.***>
보낸 날짜: Thursday, October 24, 2024 6:27:18 PM
받는 사람: humanlogio/humanlog ***@***.***>
참조: 강승곤 ***@***.***>; Author ***@***.***>
제목: Re: [humanlogio/humanlog] fix issue #65 (PR #94)


@aybabtme commented on this pull request.

________________________________

In cmd/humanlog/main_test.go<https://github.com/humanlogio/humanlog/pull/94#discussion_r1814625040>:

> @@ -0,0 +1,24 @@
+package main
+
+import (
+       "os"
+       "strings"
+       "testing"
+)
+
+func TestApplyConfigFromConfigFile_when_one_of_skip_or_keep_is_given(t *testing.T) {
+
+       wd, _ := os.Getwd()
+       dirs := strings.Split(wd, "/")


fyi: when manipulating filepath, use https://pkg.go.dev/path/filepath since it's cross platform

**Comment by KevRiver** [comment]
I rewrite the test function with T.TempDir

**Review by KevRiver — COMMENTED** [review]
@aybabtme 
I rewrote the test function with T.TempDir

**Pull request merged**